readthedocs/rtd_tests/tests/test_sync_versions.py

```diff
@@ -152,6 +153,500 @@ def test_new_tag_update_inactive(self):
+        current_stable = self.pip.get_stable_version()
+
+        # 0.8.3 is the current stable
+        self.assertEqual(
+            version8.identifier,
+            current_stable.identifier
+        )
+        self.assertTrue(current_stable.machine)
+
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+                # User new stable
+                {
+                    'identifier': 'origin/stable',
+                    'verbose_name': 'stable',
+                },
+                {
+                    'identifier': 'origin/0.8.3',
+                    'verbose_name': '0.8.3',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        version_stable = self.pip.versions.get(slug='stable')
+        current_stable = self.pip.get_stable_version()
+        self.assertEqual(
+            current_stable.identifier,
+            version_stable.identifier
+        )
+
+        # Deleting the branch should return the RTD's stable
+        version_post_data = {
+            'branches': [
+                {
+                    'identifier': 'origin/master',
+                    'verbose_name': 'master',
+                },
+                {
+                    'identifier': 'origin/0.8.3',
+                    'verbose_name': '0.8.3',
+                },
+            ],
+        }
+
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        # The version 8 should be the new stable.
+        # The stable isn't stuck with the previous branch
+        version8 = self.pip.versions.get(slug='0.8.3')
+        current_stable = self.pip.get_stable_version()
+        self.assertEqual(
+            version8.identifier,
+            current_stable.identifier,
+        )
+        self.assertNotEqual(
+            'origin/stable',
+            current_stable.identifier,
+        )
+        self.assertTrue(current_stable.machine)
+
+    @pytest.mark.xfail(strict=True)
+    def test_normal_behavior_for_stable_after_deleting_user_defined_branch_2(self):
+        """
+        The user imports a new project with a branch named ``stable``,
+        when syncing the versions, the RTD's ``stable`` is lost
+        and doesn't update automatically anymore, when the branch is deleted
+        on the user repository, the RTD's ``stable`` is back.
+        """
+        # There isn't a stable version yet
+        self.pip.versions.exclude(slug='master').delete()
```

Comment:
OK, this doesn't feel quite right. But all tests are using the pip project... Should I start using a dynamic fixture here or should I use something else? /cc @humitos 

Also, there aren't tests for new projects with already created tags/branches with latest and stable. I'm going to add them.

Reply:
You can remove that version inside your test and it won't affect the others since the fixture is loaded one each test: https://docs.djangoproject.com/en/2.0/topics/testing/tools/#topics-testing-fixtures

Reply:
Yeah, but I was saying about is that feels weird to do that setup inside the test